### PR TITLE
Add 'hide archived' setting to job list settings UI

### PR DESCRIPTION
**Pull request by rsasch**
- added support in Cromwell shim for hideArchived
- added slide toggle to settings menu in job list for 'hide archived'
- added hideArchived to local browser storage

### CODE REVIEW

File: api/jobs.yaml [hunk 1]
```diff
@@ -407,6 +407,10 @@ definitions:
         type: string
         description: >
           Returns only jobs with the specified user ID.
+      hideArchived:
+        type: string
```
Comment (calbach) [inline api/jobs.yaml]: Shouldn't this be a boolean?

File: ui/src/app/job-list/job-list.component.spec.ts [hunk 2]
```diff
@@ -44,6 +45,8 @@ describe('JobListComponent', () => {
 
   // Jobs with IDs JOB0 -> JOB4.
   function testJobs(count: number): QueryJobsResult[] {
+    const flags = ['archive','','archive','','archive'];
```
Comment (calbach) [inline ui/src/app/job-list/job-list.component.spec.ts]: opt: `new Array(3).fill('archive');`

File: ui/src/app/job-list/job-list.component.spec.ts [hunk 3]
```diff
@@ -329,6 +335,28 @@ describe('JobListComponent', () => {
     expect(de.queryAll(By.css('.fake-projects')).length).toEqual(1);
   }));
 
+  it('does not display the hide archived setting without the right project setting', async(() => {
```
Comment (calbach) [inline ui/src/app/job-list/job-list.component.spec.ts]: Nice

File: ui/src/app/job-list/job-list.component.ts [hunk 4]
```diff
@@ -71,9 +73,9 @@ export class JobListComponent implements OnInit {
     this.projectId = req.extensions.projectId || '';
 
     this.header.pageSubject.subscribe(resp => this.onClientPaginate(resp));
-    const savedPageSettings = this.settingsService.getPageSize(this.projectId);
-    if (savedPageSettings) {
-      this.pageSize = savedPageSettings;
+    this.savedProjectSettings = this.settingsService.getSettingsForProject(this.projectId);
```
Comment (calbach) [inline ui/src/app/job-list/job-list.component.ts]: Hmm, I think I probably prefer calling settingsService each time. I would worry about this local copy of the settings going out of sync over time.

File: api/jobs.yaml (hunk 1 above)
Comment (rsasch) [reply api/jobs.yaml]: fixed

File: ui/src/app/job-list/job-list.component.spec.ts (hunk 2 above)
Comment (rsasch) [reply ui/src/app/job-list/job-list.component.spec.ts]: actually, this isn't needed anymore, so I removed it altogether

File: ui/src/app/job-list/job-list.component.ts (hunk 4 above)
Comment (rsasch) [reply ui/src/app/job-list/job-list.component.ts]: ok – changed

File: ui/src/app/core/settings.service.ts [hunk 5]
```diff
@@ -43,37 +46,30 @@ export class SettingsService {
...
-    this.currentSettings.v1.projects.forEach((p) => {
-      if (p.projectId === projectId) {
-        p.displayColumns = fields;
-        return;
+  getSavedSettingValue(settingKey: string, projectId: string): any {
+    let value = null;
+    const settings = this.getSettingsForProject(projectId);
+    for (let key in settings) {
+      if (key == settingKey) {
+        value = settings[key];
```
[the remark this answers is not in the document]
Comment (rsasch) [reply ui/src/app/core/settings.service.ts]: fixed

File: ui/src/app/job-list/job-list.component.html [hunk 6]
```diff
@@ -23,6 +23,10 @@ <h2 class="settings-header">Displayed columns</h2>
                       (click)="$event.stopPropagation()">{{ df.display }}</mat-checkbox>
       </mat-list-item>
     </mat-list>
+    <div *ngIf="settingsService.getSavedSettingValue('hideArchived', projectId) != null">
```
Comment (calbach) [inline ui/src/app/job-list/job-list.component.html]: nit: I would generally consider this too heavy to put in a template, would probably extract this into a helper method on the component.

**Pull request merged**